accessories/wooxrgbwlight.coffee

```diff
@@ -0,0 +1,89 @@
+module.exports = (env) ->
+
+  hap = require 'hap-nodejs'
+  Service = hap.Service
+  Characteristic = hap.Characteristic
+
+  SwitchAccessory = require('./switch')(env)
+
+  class WooxRGBWLightAccessory extends SwitchAccessory
+    # hsb value of current color
+    _color: {
+      h: null
+      s: null
+      b: null
+    }
+    _dimlevel = 0
+    
+
+    constructor: (device) ->
+      super(device)
+      device.getHsb().then( (hsb) => 
+        @_color.h = hsb[0]
+        @_color.s = hsb[1]
+        @_color.b = hsb[2]
+      )
+      device.getDimlevel().then( (level) => @_dimlevel = level)
+      
+      @service.getCharacteristic(Characteristic.Hue)
+        .on 'get', (callback) =>
+          @handleReturnPromise(device.getHue(), callback, null)
+
+      @service.getCharacteristic(Characteristic.Hue)
+        .on 'set', (hue, callback) =>
+          if hue is @_color.h
+            callback()
+            return
+          @_color.h = hue
+          @handleVoidPromise(device.changeHueTo(hue), callback)
+
+      device.on 'hue', (hue) =>
+        console.log("Hue received from Device: #{hue}")
```

Comment:
Please use env.logger.debug for this.

Reply:
Will do, forgot to remove these. Also some more changes are coming, will submit a new PR.

Reply:
Have now checked in the accessory file with your feedback and further improvements.